src/main/java/com/bean00/RequestProcessor.java

```diff
@@ -25,20 +26,15 @@ public Response processRequest(Request request) throws IOException {
             response = new Response(Status.OK);
         } else if (!fileCanBeFound) {
             response = new Response(Status.NOT_FOUND);
+        } else if (isHEADRequest) {
+            response = new Response(Status.OK, body, false);
```

Comment:
Getting the body before you know you need it adds complexity and inefficiency to your solution. Right now, you are creating a new byte array that you either immediately overwrite or do nothing with. Then you have an extra conditional check that essentially duplicates this one.

Reply:
I'm now creating and passing the `body` locally, to where it is needed (and guaranteed to exist).